Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,294 @@ import UIKit
+                                                                color: UIColor.appColors.lightBlue)
+                                            ]
+            let defaultAttributedStringStyle = AttributedStringStyle(font: UIFont.systemFont(ofSize: 14),
+                                                                     color: UIColor.appColors.deepGray)
+            let attrMsgString =  Utils.styleAttributedString(LocalizedStrings.tapOnBlueText,
+                                                                styles: attributedStringStyles,
+                                                                defaultStyle: defaultAttributedStringStyle)
+            alertController.setValue(attrMsgString, forKey: "attributedMessage")
+
+            alertController.addAction(UIAlertAction(title: LocalizedStrings.gotIt,
                                                     style: UIAlertAction.Style.default,
                                                     handler: nil))
-            
             self.present(alertController, animated: true, completion: nil)
         }
     }
-    
-    func openLink(urlString: String) {
-        if let url = URL(string: urlString) {
-            let viewController = SFSafariViewController(url: url)
-            viewController.delegate = self as SFSafariViewControllerDelegate
-            self.navigationController?.pushViewController(viewController, animated: true)
-        }
+
+    @IBAction func showOrHideDetails(_ sender: Any) {
+        self.transactionDetailsTable.isHidden = !self.transactionDetailsTable.isHidden
+        self.showOrHideDetailsBtn.setTitle(self.transactionDetailsTable.isHidden ? LocalizedStrings.showDetails : LocalizedStrings.hideDetails, for: .normal)
     }
 }
 
 extension TransactionDetailsViewController: UITableViewDataSource, UITableViewDelegate {
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 3
+        return 4
     }
-    
+
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         return section == 0 ? self.generalTxDetails.count : 1
     }
-    
+
     func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         return 0.5
     }
-    
+
     func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
         let headerView = UIView.init(frame: CGRect.zero)
-        headerView.backgroundColor = #colorLiteral(red: 0.8039215803, green: 0.8039215803, blue: 0.8039215803, alpha: 1)
+        headerView.backgroundColor = UIColor.appColors.gray
         headerView.frame.size.height = 0
         return headerView
     }
-    
+
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         switch indexPath.section {
         case 0:
             let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionDetailCell") as! TransactionDetailCell
+            cell.presentingController = self
```

Comment:
As a follow up to earlier comments about using a callback rather than passing the entire VC:
```suggestion
            cell.onTxDetailValueCopied = { copiedDetail in
                Utils.showBanner(parentVC: self, type: .success, text: String(format: LocalizedStrings.sgCopied, copiedDetail))
            }
```